kernel/src/process/execve.rs

```diff
@@ -230,62 +230,62 @@ fn set_cpu_context(
     debug!("user stack top: 0x{:x}", elf_load_info.user_stack_top);
 }
 
-/// Sets the UID and GID in the credentials according to the ELF file.
+/// Sets the UID and GID in the credentials according to the ELF inode.
 ///
 /// The capabilities will be updated accordingly.
 fn apply_caps_from_exec(
     process: &Process,
     posix_thread: &PosixThread,
-    elf_file: &Path,
+    elf_inode: &Arc<dyn Inode>,
 ) -> Result<()> {
-    // FIXME: We need to recalculate the capabilities during execve even the executable file
+    // FIXME: We need to recalculate the capabilities during execve even the executable inode
     // does not have setuid/setgid bit.
     let credentials = posix_thread.credentials_mut();
-    set_uid_from_elf(process, &credentials, elf_file)?;
-    set_gid_from_elf(process, &credentials, elf_file)?;
+    set_uid_from_elf(process, &credentials, elf_inode)?;
+    set_gid_from_elf(process, &credentials, elf_inode)?;
     credentials.set_keep_capabilities(false);
 
     Ok(())
 }
 
-/// Sets the UID in the credentials according to the ELF file.
+/// Sets the UID in the credentials according to the ELF inode.
 ///
-/// If the ELF file has the `set_uid` bit, the effective UID is set to the same value as the ELF
-/// file's UID.
+/// If the ELF inode has the `set_uid` bit, the effective UID is set to the same value as the ELF
+/// inode's UID.
 fn set_uid_from_elf(
     current: &Process,
     credentials: &Credentials<WriteOp>,
-    elf_file: &Path,
+    elf_inode: &Arc<dyn Inode>,
```

Comment:
Actually, changes like this make me much more concerned about whether the methods in the `Inode` type are private.

I think we can do this:
```rust
// fs/utils/inode.rs:

pub trait InodePrivate {
    // private methods
}

pub trait Inode: InodePrivate {
    // public methods
}

// fs/utils/mod.rs:

pub(super) use inode::InodePrivate;
pub use inode::Inode;
```

But it's of course outside of the scope of this PR...